amd/src/courseformat/courseindex/placeholder.js

```diff
@@ -53,6 +53,7 @@ export default class Component extends BaseComponent {
         // Collect section information from the state.
         const exporter = this.reactive.getExporter();
         const data = exporter.course(state);
+        data.sections = this._nestSections(data.sections);
```

Comment:
Maybe put in a different variable, to avoid changing shared data.  (Corresponding change required in templates/courseformat/courseindex/courseindex.mustache )
```suggestion
        data.sectionsnested = this._nestSections(data.sections);
```